packages/pluggableWidgets/accordion-web/src/components/AccordionGroup.tsx

```diff
@@ -0,0 +1,50 @@
+import { createElement, ReactElement, ReactNode, useCallback, useContext, useEffect, useRef, useState } from "react";
+import { AccordionGroupsDispatch } from "./Accordion";
+
+export interface AccGroup {
+    header: ReactNode;
+    content: ReactNode;
+    collapsed?: boolean;
+    visible: boolean;
+}
+
+export interface AccordionGroupProps {
+    group: AccGroup;
+}
+
+export default function AccordionGroup(props: AccordionGroupProps): ReactElement | null {
+    const { group } = props;
+
+    const dispatch = useContext(AccordionGroupsDispatch);
```

Comment:
What is the reason we chose to use a Context for an one level deep callback? I feel like opening up an `onToggle: (isCollapsed: boolean) => void` API prop would suffice in case. Am I missing something?

Reply:
That way you can also avoid the optional chaining on the dispatch in your `toggleContentVisibility` and handle it in the parent

Reply:
Converted to a callback prop. I like your idea to create a nice `onToggle` callback prop that extracts all logic, but for that we would need to memoize functions inside the map function, because we cannot use useCallback inside map functions. This would require work, which I want to prevent for this PR. I would add it as a separate task to the board to improve it. What do you think?